include/clients/nrt/KDTreeClient.hpp

```diff
@@ -219,8 +218,9 @@ class KDTreeQuery : public FluidBaseClient, ControlIn, ControlOut
       auto  dataset = datasetClientPtr->getDataSet();
       index pointSize = dataset.pointSize();
       auto  outBuf = BufferAdaptor::Access(get<kOutputBuffer>().get());
-      index outputSize = k * pointSize;
-      if (outBuf.samps(0).size() < outputSize) return;
+      index realK = std::min(k, (outBuf.samps(0).size() / pointSize));
+      if (realK <= 0) return;
+      index outputSize =  realK * pointSize;
```

Comment:
I don't get this bit. Surely the only way to know the actual output size is to do the actual query?

Reply:
Oh, ok, I see that you're making sure that the user's buffer is fit for purpose, so you can bail if not able to take any points. Is the idea to be more permissive than the existing check? I don't `realK` is the right name, IAC, and I'm in two minds about whether it's worth doing this pre-query.